activities/activity_arielsaffer.py

```diff
@@ -0,0 +1,85 @@
+import datetime
+
+import grass.script as gs
+
+
+# compute solar irradiance (W/m2) for a given day and hour
+# using the time zone of the surface location (based on NE corner),
+# if packages are available (otherwise, current timezone is used)
+def run_radiance_now(scanned_elev, env, **kwargs):
+    try:
+        # you may need to install these packages
+        import pytz
+        from timezonefinder import TimezoneFinder
+
+        obj = TimezoneFinder()
+
+        # Get the current day/time
+        today = datetime.datetime.today()
+        # Get the lat and lon of the NE corner of the computation region
+        map_loc = gs.read_command("g.region", flags="bg", raster=scanned_elev)
+        long = float(map_loc.split()[4:6][0].split("=")[1])
+        lat = float(map_loc.split()[4:6][1].split("=")[1])
+        # Identify the time zone of the mapped region
+        map_timezone = obj.timezone_at(lng=long, lat=lat)
+        # Convert the current day/time to the time in that timezone
+        maptime = today.astimezone(pytz.timezone(map_timezone))
+        # Extract day of year and hour
+        day = maptime.timetuple().tm_yday
+        current_time = maptime.hour + maptime.minute / 60
+        print("Currently: " + str(current_time) + " hours, in Timezone " + map_timezone)
+    except Exception:
```

Comment:
Great solution overall. This is better done using ImportError instead of generic Exception. Only problems with import will cause the except branch to execute while other issues like wrong function names or some parsing issues will cause the traceback which would be the desired behavior.

Reply:
That makes sense, thanks for the pointer! I'll change that.